src/sa_web/static/js/views/legend-view.js

```diff
@@ -26,12 +26,6 @@ var Shareabouts = Shareabouts || {};
     toggleVisibility: function(evt) {
       var $cbox = $(evt.target),
         id = $cbox.attr('data-layerid');
-
-      if ($cbox.is(':checked')) {
-        $(S).trigger('visibility', [id, true]);
-      } else {
-        $(S).trigger('visibility', [id, false]);
-      }
```

Comment:
Any reason for removing this? I think this breaks our GIS functionality...

I can help test and implement these changes too